tracer/src/Datadog.Trace/Debugger/Expressions/ProbeExpressionParser.cs

```diff
@@ -399,6 +401,18 @@ private Expression HandleReturnType(Expression finalExpr)
             return finalExpr;
         }
 
+        if (typeof(T).IsNumeric()
```

Comment:
Seems a bit strange and possibly problematic that this change was made to serve Metric Probes in a probe path that serves other probe types as well. Won't this potentially have unintended side effects, like causing integers in Log Templates being rendered as floats?

Reply:
if `typeof(T)` (and not `finalExpr.Type`) is numeric it must be a metric probe. But I'm indifferent to the term probe here and refer to the method's consumer and the generic value it provided in the call.

Reply:
Right - tbh that hidden assumption is exactly what I find problematic - this is logic that we're performing doing **for** Metric Probes, but we're not explicitly calling it out nor explaining what the code is intended to do (via a comment or using a method name), so if someone in the future finds a problem with it, they might be tempted to delete it and have a hard time understanding why it's there in the first place.

If nothing else, maybe we could add a comment?

Reply:
Fixed in cbb76250b0413b828b3fadfa8c9b56355a4a8686